src/datascience-ui/ipywidgets/kernel/index.ts

```diff
@@ -220,6 +221,17 @@ function convertVSCodeOutputToExecutResultOrDisplayData(
         if (customMetadata.unconfined) {
             metadata.unconfined = customMetadata.unconfined;
         }
+    } else {
+        // New API.
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const outputMetadata = request.output.metadata as Record<string, any> | undefined;
+        if (outputMetadata && outputMetadata[request.mimeType] && outputMetadata[request.mimeType].metadata) {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            Object.assign(metadata, outputMetadata[request.mimeType].metadata);
+            if (request.mimeType in outputMetadata[request.mimeType].metadata) {
+                Object.assign(metadata, outputMetadata[request.mimeType].metadata[request.mimeType]);
```

Comment:
This seems weird? 

You assign metadata to the metadata object, and then you reassign it to just the mimetype value? Did you intend to do that?

Reply:
Yes, the VSC renderer API hasn't been updated, hence not sure what we get here, hence just sending everything (spread everything out & send it all).
I have filed an issue on VSC (and a corresponding tracking issue at our end as well) for them to update their renderer types so we know what we're getting in the request object & what needs to be send to the renderer.

Reply:
Ah okay, part of my confusion was I thought object.assign wholly replaces an object. It just assigns the fields from one object into an already existing object (leaving old fields there).

Reply:
Very Hacky, just dump everything and hope it works...